Dockerfile

```diff
@@ -35,6 +35,21 @@ RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
       --ignore-src \
     && rm -rf /var/lib/apt/lists/*
 
+#default value of arg ELASTICA_ADD when not provided in the --build-arg
+ARG ELASTICA_ADD="false"
+
+# ros2-elastica dependencies
+RUN if [ "${ELASTICA_ADD}" = "false" ]; then\
+      echo 'Not adding Elastica';\
+    else\
+      echo "Adding Elastica " &&\
+      apt-get update &&\
+      apt-get install --no-install-recommends -y tzdata dirmngr gnupg2 build-essential python3-colcon-common-extensions python3-colcon-mixin python3-rosdep python3-vcstool python3-pip  povray python3-tk &&\
```

Comment:
Have you checked if all of them are necessary or if maybe some of them are already included with the ros:foxy image we are building on?

Reply:
Some of them I'm sure of are necessary like povray, python3-tk for visualisation. Let me check the others if they are included with the base image of ros:foxy.

Reply:
I removed all of them except the last three as they were in the base image of `ros:foxy` & removing them didn't affect the original functioning